bin/po2android.js

```diff
@@ -63,79 +76,120 @@ const getUniqueName = ( function() {
+		( result, { string, stringPlural, comments } ) => {
+			if ( ! string ) {
+				return result;
+			}
+			const uniqueName = getUniqueName( string );
+			const escapedValue = androidReplacements(
+				escapeResourceXML( string )
+			);
+			const escapedValuePlural = androidReplacements(
+				escapeResourceXML( stringPlural || '' )
+			);
+			const comment = comments || '';
+			let localizedEntry = '';
+			if ( comment ) {
+				localizedEntry += `${ indent }<!-- ${ comment.replace(
+					'--',
+					'—'
+				) } -->\n`;
+			}
+			if ( stringPlural ) {
+				localizedEntry += `${ indent }<string-array name="${ uniqueName }" tools:ignore="UnusedResources">
 ${ indent }${ indent }<item>${ escapedValue }</item>
 ${ indent }${ indent }<item>${ escapedValuePlural }</item>
 ${ indent }</string-array>
 `;
-		} else {
-			localizedEntry += `${ indent }<string name="${ uniqueName }" tools:ignore="UnusedResources">${ escapedValue }</string>\n`;
-		}
-		result[ uniqueName ] = localizedEntry;
-		return result;
-	}, {} );
+			} else {
+				localizedEntry += `${ indent }<string name="${ uniqueName }" tools:ignore="UnusedResources">${ escapedValue }</string>\n`;
+			}
+			result[ uniqueName ] = localizedEntry;
+			return result;
+		},
+		{}
+	);
 	// try to minimize changes in diffs by sorting strings
 	const androidResourcesSortedList = Object.entries( androidResourcesMap )
 		.sort( ( left, right ) => left[ 0 ].localeCompare( right[ 0 ] ) )
 		.map( ( entry ) => entry[ 1 ] );
-	return `<?xml version="1.0" encoding="utf-8"?>\n<resources xmlns:tools="http://schemas.android.com/tools">\n${ androidResourcesSortedList.join( '' ) }</resources>\n`;
+	return `<?xml version="1.0" encoding="utf-8"?>\n<resources xmlns:tools="http://schemas.android.com/tools">\n${ androidResourcesSortedList.join(
+		''
+	) }</resources>\n`;
 }
 
+// Returns used strings of a specified domain
+const getUsedStrings = ( usedStringsFile ) => {
+	return require( path.resolve( usedStringsFile ) );
+};
+
 if ( require.main === module ) {
-	if ( process.stdin.isTTY ) {
-		const potFileName = process.argv[ 2 ];
-		const destination = process.argv[ 3 ];
-		const potFileContent = fs.readFileSync( potFileName );
-		const xmlOutput = po2Android( potFileContent, process.argv[ 3 ] );
-		fs.writeFileSync( destination, xmlOutput );
-	} else {
-		let inputData = '';
-		process.stdin.on( 'readable', function() {
-			const chunk = this.read();
-			if ( chunk !== null ) {
-				inputData += chunk;
-			}
-		} );
-		process.stdin.on( 'end', function() {
-			process.stdout.write( po2Android( inputData ) );
-		} );
+	const destination = process.argv[ 2 ];
+	const usedStringsFile = process.argv[ 3 ];
+
+	const usedStrings = getUsedStrings( usedStringsFile );
+	let flattenUsedStrings = Object.entries( usedStrings ).reduce(
+		( result, [ domain, strings ] ) => {
+			return [ ...result, ...Object.values( strings ) ];
+		},
+		[]
+	);
+	flattenUsedStrings = [ ...new Set( flattenUsedStrings ) ];
```

Comment:
Used strings are grouped by domain, since we only provide one localization strings files, we unify them here and their translations will be requested as part of the main apps' strings.